app/models/user.rb

```diff
@@ -0,0 +1,17 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+
+  validates :name, presence: true, length: { maximum: Settings.validation.user.size.name }
+  validates :email, presence: true, length: { maximum: Settings.validation.user.size.email},
+  format: { with: VALID_EMAIL_REGEX }, uniqueness: true
+  validates :password, presence: true, length: { minimum: Settings.validation.user.size.password}
+
+  before_save :downcase_email
+  has_secure_password
+
+  private
+
+  def downcase_email
+    before_save { email.downcase! }
```

Comment:
before save 2 lần à

Reply:
Em cho " before_save { email.downcase! } " vao instance method cho de nhin chi

Reply:
nhưng em dùng `before_save :downcase_email` ở line 9 rồi, nên method này chỉ cần nên chứa xử lý logic thôi, kh cần before_save ấy
```suggestion
    email.downcase!
```

Reply:
tiện luôn so sánh `downcase` và `downcase!`